File: features/auth/fastify/typeorm/files/src/entities/verification-token.ts
import 'reflect-metadata';
import {
  Column,
  CreateDateColumn,
  Entity,
  Index,
  PrimaryGeneratedColumn,
} from 'typeorm';

@Entity({ name: 'verification_tokens' })
@Index(['user_id', 'kind'])
export class VerificationToken {
  @PrimaryGeneratedColumn('uuid')
  id!: string;

  @Column({ name: 'user_id', type: 'uuid' })
  user_id!: string;

  @Column({ type: 'varchar', length: 32 })
  kind!: string;

  @Column({ name: 'token_hash', type: 'varchar', length: 64, unique: true })
  token_hash!: string;

  @Column({ name: 'expires_at', type: 'timestamptz' })
  expires_at!: Date;

  @Column({ name: 'consumed_at', type: 'timestamptz', nullable: true })
  consumed_at!: Date | null;

  @CreateDateColumn({ name: 'created_at', type: 'timestamptz' })
  created_at!: Date;
}

File: features/auth/go/gorm/files/internal/auth/models/verification_token.go
package authmodels

import (
	"time"

	"gorm.io/gorm"

	"projx.local/go/internal/uuid"
)

const (
	TokenKindPasswordReset = "password_reset"
	TokenKindEmailVerify   = "email_verify"
)

type VerificationToken struct {
	ID         string     `gorm:"primaryKey;type:uuid" json:"id"`
	UserID     string     `gorm:"type:uuid;index;not null" json:"user_id"`
	Kind       string     `gorm:"type:varchar(32);index;not null" json:"kind"`
	TokenHash  string     `gorm:"uniqueIndex;type:varchar(128);not null" json:"-"`
	ExpiresAt  time.Time  `gorm:"not null" json:"expires_at"`
	ConsumedAt *time.Time `json:"consumed_at,omitempty"`
	CreatedAt  time.Time  `json:"created_at"`
	UpdatedAt  time.Time  `json:"updated_at"`
}

func (VerificationToken) TableName() string { return "verification_tokens" }

func (v *VerificationToken) BeforeCreate(_ *gorm.DB) error {
	if v.ID == "" {
		v.ID = uuid.V4()
	}
	return nil
}
